### PR TITLE
Provide ScenarioMetadata for step methods

### PR DESCRIPTION
**Pull request by mchaloupka**

### CODE REVIEW

File: Examples/ByFeature/FunctionalInjection/FunctionalInjection.fsproj [hunk 1]
```diff
@@ -21,6 +21,8 @@
     <EmbeddedResource Include="ComputerHeating.feature" />
     <Compile Include="WalletSteps.fs" />
     <EmbeddedResource Include="Wallet.feature" />
+    <EmbeddedResource Include="ScenarioInformation.feature" />
```
Comment (bartelink) [inline Examples/ByFeature/FunctionalInjection/FunctionalInjection.fsproj:24]: ScenarioContext? ScenarioMeta? Is there prior art for referring to this as `Information` ?

File: README.md [hunk 2]
```diff
@@ -189,6 +189,9 @@ As shown in [Step definitions (F# without mutable field)](#step-definitions-f-wi
 
 * **Resolving dependencies:** If an instance cannot be located in the type-to-instance cache based on a preceding step having stashed the value, TickSpec will attempt to use the 'widest' constructor (the one with the most arguments) of the required type to instantiate it. Any input arguments to the constructor are all resolved recursively using the same mechanism. Any constructed instances are also cached in the type-to-instance cache, so next time it will return the same instance.
 
+* **Information about scenario:** If you want to condition your logic based on
```
Comment (bartelink) [inline README.md:192]:
```suggestion
* **accessing scenario metadata:** You can access contextual information about the scenario within which a step definition is executing (e.g. tags). To do this, add a parameter of type  `ScenarioInformation` in the step definition argument list (or in a constructor parameter list of a dependency)
```

**Review by bartelink — APPROVED** [review]
LGTM - minor suggestions for you to take/reject as desired

File: Examples/ByFeature/FunctionalInjection/FunctionalInjection.fsproj (hunk 1 above)
Comment (mchaloupka) [reply Examples/ByFeature/FunctionalInjection/FunctionalInjection.fsproj]: I like the `ScenarioMetadata`.

File: Examples/ByFeature/FunctionalInjection/ScenarioMetadataSteps.fs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module ScenarioMetadataSteps
+
+open System
+open NUnit.Framework
+open TickSpec
+
+let [<Then>] ``the test is named "(.+)"`` (testName: string) (scenarioMetadata: ScenarioMetadata) =
```
Comment (bartelink) [inline Examples/ByFeature/FunctionalInjection/ScenarioMetadataSteps.fs:7]: suggest `scenario`, `meta` or `metadata` for param name; fine with type name being explicit but it's a bit long here?

Comment (mchaloupka) [reply Examples/ByFeature/FunctionalInjection/ScenarioMetadataSteps.fs]: Sounds good to me, updated.

**Pull request merged**